core/arch/arm/kernel/tee_ta_manager.c

```diff
@@ -109,6 +109,33 @@ static void set_tee_rs(struct tee_ta_session *tee_rs)
 	thread_set_tsd(tee_rs);
 }
 
+static void tee_ta_set_busy(struct tee_ta_ctx *ctx)
+{
+	while (true) {
+		bool old_busy;
+
+		mutex_lock(&tee_ta_mutex);
+		old_busy = ctx->busy;
+		if (!old_busy)
```

Comment:
The check isn't necessary since `ctx->busy` will always be set to `true` here regardless if it has value `false` or `true` when we are calling this function, right? However, by doing so it seems like you can move stuff around to get rid of `old_busy`. The `old_busy` is used outside the mutex, so it might be that I have missed something (I haven't read all changes yet, so I might revise this comment when that has been done).